src/framework/components/system.js

```diff
@@ -50,6 +51,9 @@ class ComponentSystem extends EventHandler {
         entity[this.id] = component;
         entity.c[this.id] = component;
 
+        entity.oc.push(component);
```

Comment:
it seems you're only adding to the `oc` array, but never removing from it?

Reply:
Completely forgot about it :D